src/Adapter/Layout/QueryHandler/GetLayoutVariablesHandler.php

```diff
@@ -0,0 +1,365 @@
+            'current_tab_level' => $currentTabLevel,
+            'breadcrumbs2' => $breadcrumbs,
+            'quick_access_current_link_icon' => $quickAccessCurrentLinkIcon,
+            'quick_access_current_link_name' => $quickAccessCurrentLinkName,
+            'token' => $token,
+            'no_order_tip' => $this->getNotificationTip('order'),
+            'no_customer_tip' => $this->getNotificationTip('customer'),
+            'no_customer_message_tip' => $this->getNotificationTip('customer_message'),
+            'js_router_metadata' => $this->getRouterMetadata(),
+            'help_link' => 'https://help.prestashop.com/' . Language::getIsoById($this->context->getContext()->employee->id_lang) . '/doc/'
+                . Tools::getValue('controller') . '?version=' . _PS_VERSION_ . '&country=' . Language::getIsoById($this->context->getContext()->employee->id_lang),
+            'js_def' => Media::getJsDef(),
+            'toggle_navigation_url' => $this->context->getContext()->link->getAdminLink('AdminEmployees', true, [], [
+                'action' => 'toggleMenu',
+            ]),
+            'shop_list' => (new HelperShop())->getRenderedShopList(),
+            'js_translatable' => [],
+            'enableSidebar' => false,
+            // to be changed
+            'page_header_toolbar' => '',
+            'modal_module_list' => '',
+            'toolbar_btn' => [],
+            'headerTabContent' => '',
+            'disableDefaultErrorOutPut' => false,
+            'errors' => [],
+            'informations' => [],
+            'confirmations' => [],
+            'warnings' => [],
+            'modals' => [],
+            'url' => '',
+            'show_new_orders' => $this->configuration->get('PS_SHOW_NEW_ORDERS'),
+            'show_new_customers' => $this->configuration->get('PS_SHOW_NEW_CUSTOMERS'),
+            'show_new_messages' => $this->configuration->get('PS_SHOW_NEW_MESSAGES'),
+            'multishop_context' => false, //$this->multishop_context,
+            'default_tab_link' => $this->context->getContext()->link->getAdminLink(Tab::getClassNameById((int) $this->context->getContext()->employee->default_tab)),
+            'controller_name' => $this->controllerName,
+            'currentIndex' => $current_index, // maybe,
+            'lite_display' => $liteDisplay,
+            'display_header' => true,
+            'display_footer' => !$liteDisplay,
+            'display_header_javascript' => true,
+            // deleted:
+            //'php_errors' => [],
+            //'search_type' => Tools::getValue('bo_search_type'),
+            //'bo_query' => Tools::safeOutput(Tools::stripslashes(Tools::getValue('bo_query'))),
+        ];
+
+        if (Shop::isFeatureActive() && Shop::getTotalShops(false, null) > 1) {
+            $variables['shop_context'] = Shop::getContext();
+        }
+
+        $this->context->getContext()->controller->setMedia(true);
+        Hook::exec('actionAdminControllerSetMedia');
```

Comment:
Please use `HookManager`.

Reply:
I actually think we should _not_ use HookManager after looking at [its implementation](https://github.com/PrestaShop/PrestaShop/blob/develop/src/Adapter/HookManager.php). It looks like an attempt to build an Adapter and kindof modern version of Hook but the logic seems very flawed to me (heavily relying on `SymfonyContainer::getInstance()`). I think it seems to be an Adapter class built to be used inside legacy framework instead of being used inside Symfony pages.

Also it's only used in 2 "badly migrated" pages.

I strongly suggest we avoid using this HookManager for now. But we can start something better.